src/pages/index.js

```diff
@@ -1,130 +1,223 @@
 
   const editPopup = new PopupWithForm(
-    `#${editForm.parentElement.id}`,
+    `#${Constants.editForm.parentElement.id}`,
     handleProfileEditSubmit
   );
   const addPopup = new PopupWithForm(
-    `#${addForm.parentElement.id}`,
+    `#${Constants.addForm.parentElement.id}`,
     handleProfileAddSubmit
   );
-  const userInfo = new UserInfo(profileName, profileDescription);
 
   imagePopup.setEventListeners();
 
+  api
+    .getProfileInfo()
+    .then((results) => {
+      user.setUserInfo({ name: results.name, about: results.about });
+      user.setAvatar(results.avatar);
+    })
+    .catch((error) => {
+      console.error("Caught an error: ", error.message);
+    });
+
   //functions
+  const deletePopup = new ConfirmationPopup("#confirmation-modal");
+  deletePopup.setEventListeners();
+
+  function addCard(name, description, template, id) {
+    const items = [{ name: name, description: description }];
+    const modelCard = new Card(
+      name,
+      description,
+      template,
+      imageClickHandler,
+      (card) => {
+        deletePopup.open();
+        deletePopup.setDefaultSaveButton();
+        deletePopup.setSubmitAction(() => {
+          api
+            .deleteCard(card.id)
+            .then(() => {
+              card.deleteCard();
+            })
+            .then(() => {
+              deletePopup.updateSaveButton();
+            })
+            .then(() => {
+              deletePopup.close();
+            })
+            .catch((error) => {
+              console.error("Caught an error: ", error.message);
+            });
+        });
+      },
+      toggleLikeButton,
+      initialCardsInfo,
+      id
+    );
+    cardsSection.items = items;
+    cardsSection.addItem(modelCard.generateCard());
+  }
 
   function createCard(name, description, template) {
-    const modelCard = new Card(name, description, template, imageClickHandler);
-    cardsSection.addItem(modelCard.addCard());
+    return api
+      .addCardInfo(name, description, (name, link, id) => {
+        addCard(name, link, template, id);
+      })
+      .then(
+        addPopup.close()
+      )
+      .catch((error) => {
+        console.error("Caught an error: ", error.message);
+      })
+      .finally(() => {
+        addPopup.submitBtn.textContent = "Saving...";
+      });
   }
 
-  function handleProfileEditSubmit({ name, description }) {
-    userInfo.setUserInfo({ name: name, about: description });
-    editPopup.close();
+  function handleProfileEditSubmit(inputElements) {
+    api
+      .setUserInfo(inputElements.name, inputElements.description)
+      .then((result) => {
+        user.setUserInfo({ name: result.name, about: result.about });
+      })
+      .then(() => {
+        editPopup.close();
+      })
+      .catch((error) => {
+        console.error("Caught an error: ", error.message);
+      })
+      .finally(() => {
+        editPopup.submitBtn.textContent = "Saving...";
```

Comment:
`"Saving..."`  should be **before** you send the request to start the loading effect. And in `finally` you need to return the default text